airlock/views/request.py

```diff
@@ -1099,6 +1120,134 @@ def group_comment_visibility_public(request, request_id, group):
     return redirect(release_request.get_url(group))
 
 
+@instrument(func_attributes={"release_request": "request_id", "group": "group"})
+@require_http_methods(["POST"])
+def group_request_changes(request, request_id, group):
+    release_request = get_release_request_or_raise(request.user, request_id)
+    filegroup = release_request.filegroups.get(group)
+    if filegroup is None:
+        raise Http404(f"bad group {group}")
+
+    # Raise permission denied if user isn't allowed to review this request at
+    # this time
+    try:
+        permissions.check_user_can_currently_review_request(
+            request.user, release_request
+        )
+    except (exceptions.RequestPermissionDenied, exceptions.RequestReviewDenied) as exc:
+        raise PermissionDenied(str(exc))
+
+    # Track the number of files we update so we can report it to the user
+    approved_files = 0
+    changes_requested_already = 0
+    changes_requested = 0
+    errors = []
+    for output_file in filegroup.output_files:
```

Comment:
What do you think about moving this logic and loop into a bll method .e.g. `group_request_changes(...)` or similar?

It feels like its slightly on the edge of too much logic for a view to me? And it would push the permissions check into the BLL, which I think is the general pattern where possible, and speaks to the first couple of lines of the big block comment (although concurrency is still an issue). 

However, needing approved/changed file counts and errors does make that all a bit awkward :thinking: